### ISSUE TITLE
REPL should probably use Lua instead of yaml as output format

### ISSUE THREAD

**Issue by knazarov** [tarantool/tarantool#3834]
Currently our REPL uses yaml to output data. This is not convenient, because users can't copy-paste that data back to REPL. And it goes against the usual practice of REPL design, where REPLs use native format of the language to serialize data.

In tarantool it doesn't stop at that: if you do `box.space.foo:select()`, then de-serialize the result from yaml back to tuple, and then try to `box.space.foo:put()` the result back, you may get an error if your tuple is sparse enough to not be an array. This is due to how we serialize and de-serialize box.NULL to yaml. 

What I propose is recorded in this video:
[![asciicast](https://asciinema.org/a/213693.svg)](https://asciinema.org/a/213693)

Basically it substitutes yaml with lua, and takes proper care for showing box.NULL as it really is.

**Comment by Totktonada** [comment]
> In tarantool it doesn't stop at that: if you do box.space.foo:select(), then de-serialize the result from yaml back to tuple, and then try to box.space.foo:put() the result back, you may get an error if your tuple is sparse enough to not be an array.

I think it should work with sparse arrays (with nils) too and it is another issue.

**Comment by knazarov** [comment]
@Totktonada still, yaml output loses information about box.NULL, so it's relevant for other cases. If you make box.space.insert accept sparse arrays - fine, it checks out. But it may be important for other functions. I'm only listing insert as the most convenient example of what's broken in REPL _right now_

**Comment by Totktonada** [comment]
It seems we should use tagged value for box.NULL in yaml output.

**Comment by knazarov** [comment]
Attaching a PoC patch that can be seen in video. It's very rough and needs improvement.

https://gist.github.com/knazarov/6681c5055c1f6bf06eee8a6a09489f8b

**Comment by bofm** [comment]
Can we have trailing commas by default in multiline representation?

```lua
{
  1,
  2,  -- here the comma is added
}
```

**Comment by cyrillos** [comment]
[RFC] box/lua/console: Add console.fmt module

**Comment by cyrillos** [comment]
@knazarov Take a look please, once time permit. Comments are highly appreciated.
https://www.freelists.org/post/tarantool-patches/RFC-boxluaconsole-Add-consolefmt-module

**Comment by knazarov** [comment]
From the code I didn't understand if it's enabled by default or not. If it requires a configuration flip, then there is little use for it, because few people will ever get to the point of knowing this exists, let alone switching it on for all installations.

**Comment by cyrillos** [comment]
For now it is not enabled by default. I did so for backward compatibility sake. Kostya said for 2.x series we should use Lua format by default, so I will rework this patch.

**Comment by kyukhin** [comment]
Done w/ af9ef21
